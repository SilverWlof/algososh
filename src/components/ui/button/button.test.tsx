import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import { Button } from './button';

// 1 button with text
it('button with text', () => {
    const tree = renderer
        .create(<Button text="button test 2 text" />)
        .toJSON();
    expect(tree).toMatchSnapshot();
}); 

// 2 button without test
it('button without test', () => {
    const tree = renderer
        .create(<Button />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

// 3 blocked button
it('blocked button', () => {
    const tree = renderer
        .create(<Button disabled={true} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

// 4 button with loader
it('button with loader', () => {
    const tree = renderer
        .create(<Button isLoader={true} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
// 5 button handler
it('button handler', () => {
    window.alert = jest.fn();

    const onClickTest = () => {
        alert('ClickHandler')
    }

    render(<Button text="button handle click test" onClick={onClickTest} />)

    const button = screen.getByText("button handle click test");

    fireEvent.click(button);

    expect(window.alert).toHaveBeenCalledWith('ClickHandler');
}); 