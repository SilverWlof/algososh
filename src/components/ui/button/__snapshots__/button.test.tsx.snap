// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blocked button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`button with loader 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    button test 2 text
  </p>
</button>
`;

exports[`button without test 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;